diff --git a/third_party/dirent_win32.h b/third_party/dirent_win32.h
index 9b8f1d30..e459f904
--- a/third_party/dirent_win32.h
+++ b/third_party/dirent_win32.h
@@ -531,14 +531,19 @@ opendir(
         return NULL;
     }
 
+    if (strlen(dirname) > PATH_MAX) {
+        dirent_set_errno (ENAMETOOLONG);
+        return NULL;
+    }
+
     /* Allocate memory for DIR structure */
     dirp = (DIR*) malloc (sizeof (struct DIR));
     if (dirp) {
-        wchar_t wname[PATH_MAX];
+        wchar_t wname[PATH_MAX + 1];
         size_t n;
 
         /* Convert directory name to wide-character string */
-        error = dirent_mbstowcs_s (&n, wname, PATH_MAX, dirname, PATH_MAX);
+        error = dirent_mbstowcs_s (&n, wname, PATH_MAX + 1, dirname, PATH_MAX + 1);
         if (!error) {
 
             /* Open directory stream using wide-character name */
